lxd/network/openvswitch/ovn.go

```diff
@@ -2047,3 +2082,192 @@ func (o *OVN) LogicalRouterPeeringDelete(opts OVNRouterPeering) error {
 
 	return nil
 }
+
+// GetHardwareAddress gets the mac address of the ovn managed network.
+func (o *OVN) GetHardwareAddress(networkID int64) (string, error) {
+	nameFilter := fmt.Sprintf("name=lxd-net%d-lr-lrp-ext", networkID)
+	data, err := o.nbctl("--data=bare", "--format=csv", "find", "Logical_Router_Port", nameFilter)
+	if err != nil {
+		return "", err
+	}
+
+	rows, err := readCSV(data)
+	if err != nil {
+		return "", err
+	}
+
+	if len(rows) != 1 || rows[0]["mac"] == "" {
+		return "", fmt.Errorf("couldn't get hardware address")
+	}
+
+	return rows[0]["mac"], nil
+}
+
+// GetMTU gets the mtu of the ovn managed network.
+func (o *OVN) GetMTU(networkID int64) (int, error) {
+	nameFilter := fmt.Sprintf("name=lxd-net%d-lr-lrp-int", networkID)
+	data, err := o.nbctl("--data=bare", "--format=csv", "find", "Logical_Router_Port", nameFilter)
+	if err != nil {
+		return 0, err
+	}
+
+	rows, err := readCSV(data)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(rows) != 1 || rows[0]["ipv6_ra_configs"] == "" {
+		return 0, fmt.Errorf("couldn't get mtu")
+	}
+
+	kvPairs := strings.Fields(rows[0]["ipv6_ra_configs"])
+	for _, kvPair := range kvPairs {
+		kv := strings.Split(kvPair, "=")
+		if len(kv) == 2 && kv[0] == "mtu" {
+			return strconv.Atoi(kv[1])
+		}
+	}
+
+	return 0, fmt.Errorf("couldn't find mtu")
+}
+
+// GetChassis gets the name of the ovn chassis (host+gateway)
+func (o *OVN) GetChassis(networkID int64) (string, error) {
+	filter := fmt.Sprintf("logical_port=cr-lxd-net%d-lr-lrp-ext", networkID)
+	data, err := o.sbctl("--data=bare", "--format=csv", "find", "Port_Binding", filter)
+	if err != nil {
+		return "", err
+	}
+
+	rows, err := readCSV(data)
+	if err != nil {
+		return "", err
+	}
+
+	if len(rows) != 1 || rows[0]["chassis"] == "" {
+		return "", fmt.Errorf("couldn't get chassis")
```

Comment:
Upper case first char of error

Reply:
I tend to used Failed getting or Error getting rather than "couldn't" to aid in language conversion.